### PR TITLE
add linker -wrap flag

### PR DESCRIPTION
**Pull request by zachcheu**
Fixes: #15051

This passes `-wrap` through to the old.

Test
```
> cat wrap_example.c foo.c
#include <stdio.h>
extern int foo();
extern int __real_foo();

int __wrap_foo() {
    printf("wrap foo\n");
    return 0;
}

int __real_foo(){
    printf("real foo\n");
    return 0;
}

int main () {
    printf("foo:");foo();
    printf("wrapfoo:");__wrap_foo();
    printf("realfoo:");__real_foo();
    return 0;
}
#include <stdio.h>
int foo() {
    printf("foo\n");
    return 0;
}

> ./zig2 cc wrap_example.c foo.c  -Wl,-wrap,foo -o main

> ./main
foo:wrap foo
wrapfoo:wrap foo
realfoo:foo
```

### CODE REVIEW

**Review by motiejus — APPROVED** [review]
cc @kubkon 

This is causing annoyance in Uber's Go Monorepo. If you agree this is a simple-enough change, can we get it landed?

File: src/main.zig [hunk 1]
```diff
@@ -2263,6 +2267,14 @@ fn buildOutputType(
                             next_arg,
                         });
                     };
+                } else if (mem.eql(u8, arg, "-wrap") or
+                    mem.eql(u8, arg, "--wrap"))
```
[the remark this answers is not in the document]
Comment (kubkon) [reply src/main.zig]: Good catch about `-Wl,-wrap` being implicitly handled in the loop - I missed that.

Re the latter, `zig build-exe` does not understand `-Wl` flag and should handle an explicit `-wrap` flag.

**Review by kubkon — CHANGES_REQUESTED** [review]

File: src/main.zig (hunk 1 above)
[the remark this answers is not in the document]
Comment (zachcheu) [reply src/main.zig]: Is this what you mean by supporting `zig build-exe`?
```
> ./zig2 build-exe -lc main.c foo.c -wrap foo

> ./main
foo:wrap foo
wrapfoo:wrap foo
realfoo:foo
```

**Comment by andrewrk** [comment]
hi @zachcheu, I'm sorry for confusing you - I added a commit to the branch and force pushed, and then set to auto-merge once the CI tests passed. It looks like you attempted to push some other work after that just now, but ran into git troubles.

I can take it from here. Are you OK with that?

**Comment by zachcheu** [comment]
No worries. Feel free to take it from here. Thanks!

**Pull request merged**